src/main/java/com/mock/preparation/ds/Factorial.java

```diff
@@ -0,0 +1,46 @@
+package com.mock.preparation.ds;
+
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+/*
+class to find  a factorial of given number. There 2 implementation
+1. using loop
+ 2- using recursion
+ */
+public class Factorial {
+    public static void main(String[] args) {
+        try {
+            Scanner scanner = new Scanner(System.in);
+            System.out.println("Enter the no for which you want to compute the factorial");
+            int num = scanner.nextInt();
+
+            //using loop
+          int factorial1 =   factorialUsingLoop(num);
+            System.out.println("factorial of given number: "+num+" using loop is: "+factorial1);
+
+            //using recursion
+           int factorial2 = factorialUsingRecursion(num);
+            System.out.println("factorial of given number: "+num+" using recursion is: "+factorial1);
```

Comment:
you are using variable factorial1. it should be factorial2 which is computed from using recorsion method